### ISSUE TITLE
Handle run time errors in a more flexible way by putting them in en.error

### ISSUE THREAD

**Issue by Nava2** [umple/umple#279]
Originally reported on Google Code with ID 380

---

Currently in several places there are cases where RuntimeExceptions are thrown (in each
supported language). 

For example:
- There is an attempt to construct an object which would result in immediate violation
  of an invariant (as in issue 187).
- Or if you have a 1--1 association

class X {  1 -- 1 Y;}
class Y {}

There may be other cases, so a thorough search should be done

---

Reported by @umple on 2013-03-04 16:10:22

**Comment by Nava2** [comment]
Reported by @umple on 2013-09-17 13:17:35
- **Owner removed**

**Comment by Nava2** [comment]
Reported by @umple on 2013-09-17 13:26:30
- **Status changed**: `Accepted`

**Comment by Nava2** [comment]
Reported by @umple on 2014-09-05 14:52:32
- **Labels added**: ucosp

**Comment by vahdat-ab** [comment]
@TimLethbridge I couldnot understand this issue.
Looks like it's been fixed.

**Comment by TimLethbridge** [comment]
No, this has not been fixed. The idea is that there are number of runtime errors that occur, e.g. when contracting in cases where it is not possible. The text of the message is hard coded. The idea would be to soft-code them so they could be translated (en.error for English, and fr.error for French and so on). The plan is to create language translation files for all error messages anyway

> On Jan 16, 2016, at 10:54, Vahdat Abdelzad notifications@github.com wrote:
> 
> @TimLethbridge https://github.com/TimLethbridge I couldnot understand this issue.
> Looks like it's been fixed.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/umple/umple/issues/279#issuecomment-172217987.

Timothy C. Lethbridge, PhD, P.Eng., I.S.P., CSDP
Professor of Software Engineering and Computer Science
Faculté de genie / Faculty of Engineering
Université d'Ottawa / University of Ottawa
Tel: 613-562-5800x6685   Fax: 613-562-5664    Mobile: 613-252-1850
http://www.site.uottawa.ca/~tcl

**Comment by vahdat-ab** [comment]
Thanks.
